.github/workflows/publish-pypi.yml

```diff
@@ -18,35 +18,17 @@ jobs:
           TAG: ${{ github.event.release.tag_name }}
         run: echo "VERSION=${TAG#v}" >> $GITHUB_OUTPUT
 
-      - name: Set up Python 3.9
-        uses: actions/setup-python@v3
-        with:
-          python-version: 3.9
-
-      - name: Cache Poetry virtualenv
-        uses: actions/cache@v3
-        id: cache
-        with:
-          path: ~/.virtualenvs
-          key: poetry-${{ hashFiles('**/poetry.lock') }}
-          restore-keys: |
-            poetry-${{ hashFiles('**/poetry.lock') }}
+      - name: Install Poetry
+        run: pipx install poetry
 
-      - name: Install poetry
-        uses: snok/install-poetry@v1
+      - name: Set up Python 3.12
```

Comment:
we don't to support python 2.7?

Reply:
For a long while now. it is also dead for years.
Python 3.9 will reach EOL in less than a year so I took the opportunity to bump the version here